atom/browser/autofill/atom_autofill_client.cc

```diff
@@ -0,0 +1,247 @@
+#include "base/bind.h"
+#include "base/command_line.h"
+#include "base/logging.h"
+#include "base/memory/ptr_util.h"
+#include "base/strings/utf_string_conversions.h"
+#include "brave/browser/brave_browser_context.h"
+#include "build/build_config.h"
+#include "chrome/browser/profiles/incognito_helpers.h"
+#include "components/autofill/content/browser/content_autofill_driver.h"
+#include "components/autofill/content/browser/content_autofill_driver_factory.h"
+#include "components/autofill/core/browser/popup_item_ids.h"
+#include "components/autofill/core/browser/webdata/autofill_webdata_service.h"
+#include "components/autofill/core/common/autofill_pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/common/origin_util.h"
+#include "native_mate/dictionary.h"
+
+DEFINE_WEB_CONTENTS_USER_DATA_KEY(autofill::AtomAutofillClient);
+
+// stubs TODO - move to separate files
+#include "net/http/http_request_headers.h"
+namespace variations {
+void AppendVariationHeaders(const GURL& url,
+                            bool incognito,
+                            bool uma_enabled,
+                            net::HttpRequestHeaders* headers) {
+}
+}
+
+namespace rappor {
+void SampleDomainAndRegistryFromGURL(RapporService* rappor_service,
+                                     const std::string& metric,
+                                     const GURL& gurl) {}
+}  // namespace rappor
+// end stubs
+
+namespace mate {
+
+template<>
+struct Converter<autofill::Suggestion> {
+  static v8::Local<v8::Value> ToV8(
+    v8::Isolate* isolate, autofill::Suggestion val) {
+    mate::Dictionary dict = mate::Dictionary::CreateEmpty(isolate);
+    dict.Set("value", val.value);
+    dict.Set("frontend_id", val.frontend_id);
+    return dict.GetHandle();
+  }
+};
+}  // namespace mate
+
+namespace autofill {
+
+AtomAutofillClient::AtomAutofillClient(content::WebContents* web_contents)
+    : content::WebContentsObserver(web_contents),
+      api_web_contents_(nullptr) {
+  DCHECK(web_contents);
+}
+
+AtomAutofillClient::~AtomAutofillClient() {
+}
+
+void AtomAutofillClient::Initialize(atom::api::WebContents* api_web_contents) {
+  api_web_contents_ = api_web_contents;
+}
+
+void AtomAutofillClient::DidAcceptSuggestion(const std::string& value,
+                                             int frontend_id,
+                                             int index) {
+  if (delegate_) {
+    delegate_->DidAcceptSuggestion(base::UTF8ToUTF16(value), frontend_id,
+                                   index);
+  }
+}
+
+PersonalDataManager* AtomAutofillClient::GetPersonalDataManager() {
+  content::BrowserContext* context = web_contents()->GetBrowserContext();
+  return PersonalDataManagerFactory::GetForBrowserContext(
+        chrome::GetBrowserContextRedirectedInIncognito(context));
```

Comment:
you don't need `GetBrowserContextRedirectedInIncognito` here because the factory already handles that